nano/node/wallet.cpp

```diff
@@ -1473,15 +1473,15 @@ void nano::work_watcher::run ()
 				root = i.second->root ();
 				nano::work_validate (root, i.second->block_work (), &difficulty);
 			}
-			if (node.active.active_difficulty () > difficulty && i.second != nullptr)
+			if (node.active.limited_active_difficulty () > difficulty && i.second != nullptr)
 			{
 				auto qualified_root = i.second->qualified_root ();
 				auto hash = i.second->hash ();
 				nano::state_block_builder builder;
 				std::error_code ec;
 				builder.from (*i.second);
 				lock.unlock ();
-				builder.work (node.work_generate_blocking (root, node.active.active_difficulty ()));
+				builder.work (node.work_generate_blocking (root, node.active.limited_active_difficulty ()));
```

Comment:
Seems like `active_difficulty` could change in between these two calls

Reply:
Right